hypothesis-python/tests/conftest.py

```diff
@@ -76,3 +78,30 @@ def freeze():
         assert sys.version_info[0] == 2
     monkeypatch.setattr(time_module, "sleep", sleep)
     monkeypatch.setattr(time_module, "freeze", freeze, raising=False)
+
+
+random_states_after_tests = {}
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_call(item):
+    # This hookwrapper checks for PRNG state leaks from Hypothesis tests.
+    # See: https://github.com/HypothesisWorks/hypothesis/issues/1919
+    if not (hasattr(item, "obj") and is_hypothesis_test(item.obj)):
+        yield
+    else:
+        # We start by peturbing the state of the PRNG, because repeatedly
+        # leaking PRNG state resets state_after to the (previously leaked)
+        # state_before, and that just shows as "no use of random".
+        random.seed(random.randrange(2 ** 32))
```

Comment:
We still need the change from the other PR to use a dedicated Random obejct for this, right? Otherwise seeding the PRNG consistently will still result in this new seed being equally consistent.